### PR TITLE
File limits

### PR DESCRIPTION
**Pull request by jvehent**
These patches implement the limit of walking errors returned by the file module and address a few other minor things.

r? @ameihm0912

### CODE REVIEW

File: modules/file/file.go [hunk 1]
```diff
@@ -1618,7 +1626,14 @@ func (r *run) buildResults(t0 time.Time) (resStr string, err error) {
 				}
 				mf.Search = search
 				mf.Search.Options.MatchLimit = 0
+				// store the value of maxerrors if greater than the one
+				// we already have, we'll need it further down to return
+				// the right number of walking errors
+				if int(mf.Search.Options.MaxDepth) > maxerrors {
```
Comment (ameihm0912) [inline modules/file/file.go]: Should these refer to MaxErrors and not MaxDepth?

File: modules/file/file.go [hunk 2]
```diff
@@ -1649,7 +1664,14 @@ func (r *run) buildResults(t0 time.Time) (resStr string, err error) {
 					mf.Search.Paths = search.Paths
 				}
 				mf.Search.Options.MatchLimit = 0
+				// store the value of maxerrors if greater than the one
+				// we already have, we'll need it further down to return
+				// the right number of walking errors
+				if int(mf.Search.Options.MaxDepth) > maxerrors {
```
Comment (ameihm0912) [inline modules/file/file.go]: Similar here

File: modules/file/file.go (hunk 1 above)
Comment (jvehent) [inline modules/file/file.go]: Good catch! Copy/paste ftw :facepunch:

**Comment by ameihm0912** [comment]
r+ other than that one noted issue

**Pull request merged**